custom_components/iotawatt/sensor.py

```diff
@@ -1,137 +1,266 @@
         )
-        entities = [entity]
-        async_add_entities(entities)
 
-    async_dispatcher_connect(hass, SIGNAL_ADD_DEVICE, async_new_entities)
+    async_add_entities(_create_entity(key) for key in coordinator.data["sensors"])
 
+    @callback
+    def new_data_received():
+        """Check for new sensors."""
+        entities = [
+            _create_entity(key)
+            for key in coordinator.data["sensors"]
+            if key not in created
+        ]
+        if entities:
+            async_add_entities(entities)
 
-class IotaWattSensor(IotaWattEntity):
-    """Defines a IoTaWatt Energy Sensor."""
+    coordinator.async_add_listener(new_data_received)
 
-    def __init__(self, coordinator, entity, mac_address, name):
-        """Initialize the sensor."""
-        super().__init__(
-            coordinator=coordinator, entity=entity, mac_address=mac_address, name=name
-        )
 
-        sensor = self.coordinator.data["sensors"][entity]
-        self._ent = entity
-        self._name = name
-        self._io_type = sensor.getType()
-        self._state = None
-        self._attr_state_class = STATE_CLASS_MEASUREMENT
-        self._attr_force_update = True
-
-        unit = sensor.getUnit()
-        if unit == "Watts":
-            self._attr_unit_of_measurement = POWER_WATT
-            self._attr_device_class = DEVICE_CLASS_POWER
-        elif unit == "WattHours":
-            self._attr_unit_of_measurement = ENERGY_WATT_HOUR
-            self._attr_device_class = DEVICE_CLASS_ENERGY
-        elif unit == "Volts":
-            self._attr_unit_of_measurement = ELECTRIC_POTENTIAL_VOLT
-            self._attr_device_class = DEVICE_CLASS_VOLTAGE
-        else:
-            self._attr_unit_of_measurement = unit
+class IotaWattSensor(update_coordinator.CoordinatorEntity, RestoreEntity, SensorEntity):
+    """Defines a IoTaWatt Energy Sensor."""
 
-    @property
-    def device_state_attributes(self):
-        """Return the state attributes of the device."""
-        if self._io_type == "Input":
-            channel = self.coordinator.data["sensors"][self._ent].getChannel()
-        else:
-            channel = "N/A"
+    entity_description: IotaWattSensorEntityDescription
+    _attr_force_update = True
 
-        attrs = {"type": self._io_type, "channel": channel}
+    def __init__(
+        self,
+        coordinator,
+        key,
+        entity_description: IotaWattSensorEntityDescription,
+    ):
+        """Initialize the sensor."""
+        super().__init__(coordinator=coordinator)
 
-        return attrs
+        self._key = key
+        data = self._sensor_data
+        self._accumulating = data.getUnit() == "WattHours" and not data.getFromStart()
+        self._accumulated_value = None
+        if data.getType() == "Input":
+            unit = data.getUnit() + self._name_suffix
+            self._attr_unique_id = (
+                f"{data.hub_mac_address}-input-{data.getChannel()}-{unit}"
+            )
```

Comment:
```suggestion
        if data.getType() == "Input":
            unit = data.getUnit() + self._name_suffix
            self._attr_unique_id = (
                f"{data.hub_mac_address}-input-{data.getChannel()}-{unit}"
            )
        elif data.getType() == "Output":
            self._attr_unique_id = (
                f"{data.hub_mac_address}-output-{data.getName()}-{data.getUnit()}"
            ) 
```

Reply:
Doing that would break transitioning to HA iotawatt integration though.

Reply:
ok don't add it